core/ustring.cpp

```diff
@@ -788,6 +788,57 @@ Vector<String> String::split_spaces() const {
 	return ret;
 }
 
+Vector<String> String::split_lines(bool p_allow_empty, int p_maxsplit) const {
+	Vector<String> ret;
+
+	if (length() > 0) {
+
+		int curr_line_begin = 0;
+
+		CharType curr_char;
+		CharType next_char = get(0);
+
+		for (int i = 0; i < length(); i++) {
+			curr_char = next_char;
+			next_char = get(i + 1); // will get the null terminator when i == length() - 1
+
+			auto advance = [&](CharType twin) {
```

Comment:
I am not sure this is an acceptable implementation regarding our coding standards. We do not use that kind of modern C++ features like auto, as they hurt the code readability a lot.

I'm not familiar with modern C++ feature, so I can't tell if there's another way to implement that. But I think avoiding to use auto and such iterators would likely be better.

Reply:
Indeed, you'll have to remove the use of `auto` and replace lambdas/iterators with traditional loops.

Reply:
Understood. I was trying to reduce code duplication and still handle both `/r/n` and `/n/r` *just in case* they appeared in that order ([Wikipedia](https://en.wikipedia.org/wiki/Newline?wprov=sfla1) says that `/n/r` is only used on Acorn BBC and RISC OS, but maybe there could be some weird endianness issue somewhere?). I figured *if* they appeared in that order, it was probably best to treat them as a single newline, but looking at other places in the code, it seems Godot usually ignores `/n/r` anyway, so maybe I should just not handle that case and let them be 2 separate newlines?